client/landing/stepper/declarative-flow/site-migration-flow.ts

```diff
@@ -30,12 +31,20 @@ const FLOW_NAME = 'site-migration';
 const siteMigration: Flow = {
 	name: FLOW_NAME,
 	isSignupFlow: false,
+	__experimentalUseSessions: true,
 
 	useSideEffect() {
 		const { setIntent } = useDispatch( ONBOARD_STORE );
 		useEffect( () => {
 			setIntent( Onboard.SiteIntent.SiteMigration );
 		}, [] );
+		const { set, get } = useFlowState();
+		const urlQueryParams = useQuery();
+		const ref = urlQueryParams.get( 'ref' );
+
+		if ( ref && ! get( 'migration' )?.entryPoint ) {
```

Comment:
The ref is used to identify how the user arrives on the site setup flow. 
It is stored on the session to be reused across flows.

Reply:
That's a great idea! Maybe record the ref [in the hook](https://github.com/Automattic/wp-calypso/blob/422fc1fb19c8acb271f8e008df99880e58365620/client/landing/stepper/declarative-flow/internals/state-manager/store.ts#L13-L25)? Seems like a generally useful piece of info. Maybe use `initialData` to record the ref. Like so

```
const PERSISTENCE_CONFIG = {
	staleTime: Infinity,
	refetchOnMount: false,
	refetchOnWindowFocus: false,
	refetchOnReconnect: false,
	/**
	 * This query doesn't really depend on network, so ask useQuery to always fetch even if the network is offline.
	 */
	networkMode: 'always',
	initialData: {
		ref: 'hello',
	}
} as const;
```

Reply:
It could be a future improvement as soon we see it working fine. 
I would like to see it in production first before optimizing.